Client.UnitTests/JobWorkerTest.cs

```diff
@@ -659,6 +660,44 @@ public void ShouldUseAutoCompleteWithWorker()
         Assert.Contains(3, completeJobRequests);
     }
 
+    [Test]
+    public async Task CanDisposeRunningWorker()
+    {
+        // given
+        TestService.AddRequestHandler(
+            typeof(ActivateJobsRequest),
+            _ => CreateExpectedResponse());
+        var signal = new EventWaitHandle(false, EventResetMode.AutoReset);
+
+        // when
+        var jobWorker = ZeebeClient.NewWorker()
+            .JobType("foo")
+            .Handler(async (jobClient, job) =>
+            {
+                // trigger worker disposal with second job
+                signal.Set();
+
+                // this should not be completed
+                await jobClient.NewCompleteJobCommand(job).Send();
```

Comment:
I feel like this has high potential for a flaky test. While I like that we have a test for this, not sure whether it is valuable to have a flaky one.